src/main/scala/sparkz/core/app/Application.scala

```diff
@@ -67,14 +68,23 @@ trait Application extends ScorexLogging {
     settings.network.declaredAddress
   }
 
-  val sparkzContext = SparkzContext(
+  val sparkzContext: SparkzContext = SparkzContext(
     messageSpecs = basicSpecs ++ additionalMessageSpecs,
     features = features,
     timeProvider = timeProvider,
     externalNodeAddress = externalSocketAddress
   )
 
-  val peerManagerRef = PeerManagerRef(settings, sparkzContext)
+  private val secureRandom = new SecureRandom()
+  private val nKey: Int = secureRandom.nextInt()
+  private val newBucketConfig: BucketConfig = BucketConfig(buckets = 1024, bucketPositions = 64, bucketSubgroups = 64)
+  private val triedBucketConfig: BucketConfig = BucketConfig(buckets = 256, bucketPositions = 64, bucketSubgroups = 8)
+  private val triedBucket: TriedPeerBucketStorage = TriedPeerBucketStorage(triedBucketConfig, nKey, timeProvider)
+  private val newBucket: NewPeerBucketStorage = NewPeerBucketStorage(newBucketConfig, nKey, timeProvider)
+
+  private val bucketManager: BucketManager = new BucketManager(newBucket, triedBucket)
+  private val peerDatabase = new InMemoryPeerDatabase(settings.network, sparkzContext.timeProvider, bucketManager)
+  val peerManagerRef: ActorRef = PeerManagerRef(settings, sparkzContext, peerDatabase)
```

Comment:
what do you think about moving this initialization vars into a method returning `peerManagerRef` in the end?

Reply:
Sounds good